core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -147,11 +147,8 @@ trait EntityEncoderInstances extends EntityEncoderInstances0 {
   implicit val byteArrayEncoder: EntityEncoder[Array[Byte]] =
     chunkEncoder.contramap(Chunk.bytes)
 
-  // TODO fs2 port this is where we miss ByteVector
-  /*
   implicit val byteBufferEncoder: EntityEncoder[ByteBuffer] =
-    chunkEncoder.contramap(ByteVector.view)
-   */
+    chunkEncoder.contramap(bb => Chunk.bytes(bb.array))
```

Comment:
After reading a bit more you're correct, `array` is optional. It seems to me that the totally safe way is to create a `Array[Byte]` and copy the data using `get`. This is guaranteed to work and it will separate the `array` on the buffer from the one we put in the `Chunk`

It maybe terrible performance-wise though

We could check the `hasArray` method to decide to do it the other way, but changes to the original `ByteBuffer` will be reflected on the `Chunk`. scodec comment indicate that the caller should be careful about not changing the original buffer